File: lib/domain/entities/ai_insight.dart
class AiInsightEntity {
  final String id;
  final String category;
  final String severity;
  final String title;
  final String body;
  final List<String> actionItems;
  final bool isRead;
  final bool isDismissed;
  final DateTime createdAt;

  const AiInsightEntity({
    required this.id,
    required this.category,
    required this.severity,
    required this.title,
    required this.body,
    required this.actionItems,
    required this.isRead,
    required this.isDismissed,
    required this.createdAt,
  });

  bool get isCritical => severity == 'critical';
  bool get isWarning => severity == 'warning';
}

class WhyLosingIssue {
  final String issue;
  final String severity;
  final String evidence;
  final double? impactUsd;
  final String explanation;

  const WhyLosingIssue({
    required this.issue,
    required this.severity,
    required this.evidence,
    this.impactUsd,
    required this.explanation,
  });
}

class BehavioralMistake {
  final String mistake;
  final int count;
  final double? estimatedLossUsd;
  final String howToFix;

  const BehavioralMistake({
    required this.mistake,
    required this.count,
    this.estimatedLossUsd,
    required this.howToFix,
  });
}

class WhyLosingReport {
  final String overallAssessment;
  final List<WhyLosingIssue> primaryIssues;
  final List<BehavioralMistake> behavioralMistakes;
  final List<String> actionableSteps;
  final double? estimatedRecoverableLoss;
  final DateTime generatedAt;

  const WhyLosingReport({
    required this.overallAssessment,
    required this.primaryIssues,
    required this.behavioralMistakes,
    required this.actionableSteps,
    this.estimatedRecoverableLoss,
    required this.generatedAt,
  });
}

File: backend/app/schemas/ai_insight.py
from datetime import datetime

from pydantic import BaseModel


class AiInsightResponse(BaseModel):
    id: str
    category: str
    severity: str
    title: str
    body: str
    action_items: list[str]
    is_read: bool
    is_dismissed: bool
    expires_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}


class WhyLosingMoneyReport(BaseModel):
    overall_assessment: str
    primary_issues: list[dict]
    behavioral_mistakes: list[dict]
    structural_problems: list[dict]
    actionable_steps: list[str]
    estimated_recoverable_loss_usd: float | None
    generated_at: datetime


class MarkInsightReadRequest(BaseModel):
    insight_ids: list[str]
